### ISSUE TITLE
dbus-systemcalc-py: Make multiple BMSes in the same system work well

### ISSUE THREAD

**Issue by izak** [victronenergy/venus#420]
Current situation: DVCC will use the first seen battery service with BMS paths to control the solar chargers and the Multi. In most systems this is fine, as there is only one BMS. However; as soon as you have a system with two batteries banks (for example one Lynx Ion BMS + HE batteries with Quattros; and another one used for a bow thruster); this no longer works: the `first seen` part introduces randomness.

The upside of doing that, rather than just taking the main battery service as auto- or manually selected by the user, is that systems with for example a BYD and a BMV in parallel are possible: the BMV is used as the battery monitor; while DVCC takes its data from the BYD service.

Desired situation:
- make the used battery service for DVCC predictable; by using device instance.
- do that without breaking the BYD + BMV use case.
- supporting parallel or even parallel-redundant battery systems from within dbus-systemcalc/dvcc is not in scope; nor wanted in the future. It might be done; but then outside of systemcalc or even outside of Venus OS.

The spec of the new implementation is:
- If the active battery monitor has BMS paths, use that for DVCC.
- Else, look for other battery services that have those paths; sorted by instance; and use the one with the lowest instance.

**Comment by mpvader** [comment]
Hi Izak,

this is technically nice; but I don't see a reason to work on this from software. My point of view:

- Using a BMV with a BYD: if that is an issue; we officially state that its not supported and issue closed.
- Supporting multiple BMS-es: I don't want systemcalc to have to add up the allowed charge currents of different BMSes; and-so-forth.

So closing; welcome to discuss or re-open if I overlooked something.

**Comment by mpvader** [comment]
Hi Izak, you wrote that DVCC will now use the first available battery service with BMS paths.

What sorting is used? Or is it sorted by time of discovery?

In the code I see this:
https://github.com/victronenergy/dbus-systemcalc-py/blob/master/delegates/dvcc.py#L404

**Comment by mpvader** [comment]
I've updated the initial tekst to the new spec. Izak will work on that soon. (within days)

**Comment by mpvader** [comment]
Added in v2.30~29; with a followup bugfix in v2.30~30
